### PR TITLE
Fixes issue #757

### PR DESCRIPTION
**Pull request by fulv**

### CODE REVIEW

File: src/collective/cover/profiles/testfixture/structure/.objects [hunk 1]
```diff
@@ -1,3 +1,4 @@
 my-image,Image
 my-image1,Image
 my-image2,Image
+my-image3,Image
```
Comment (djowett) [inline src/collective/cover/profiles/testfixture/structure/.objects]: I'm not sure you need this extra image - and it seems to be breaking the robot tests a bit more than usual!
https://travis-ci.org/collective/collective.cover/jobs/284702469

File: src/collective/cover/widgets/textlinessortable.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def get_custom_title(self, uuid):
             return title
         # If didn't find, get object title
         obj = uuidToObject(uuid)
-        return obj.Title()
+        return obj.Title().decode('utf-8')
```
Comment (djowett) [inline src/collective/cover/widgets/textlinessortable.py]: is using `safe_unicode()` not, well, safer?  I think (but not sure) there's a danger that obj.Title() may be e.g. u'\xc6\xb5' (as opposed to '\xc6\xb5') in which case using `decode('utf-8')` will break.

File: src/collective/cover/widgets/textlinessortable.py [hunk 3]
```diff
@@ -99,7 +99,7 @@ def get_custom_description(self, uuid):
             return description
         # If didn't find, get object description
         obj = uuidToObject(uuid)
-        return obj.Description()
+        return obj.Description().decode('utf-8')
```
Comment (djowett) [inline src/collective/cover/widgets/textlinessortable.py]: Again, question if using safe_unicode() is safer

File: src/collective/cover/tests/test_textlinessortable_widget.py [hunk 4]
```diff
@@ -158,20 +158,65 @@ def test_extract(self):
         self.assertDictEqual(extracted_value, expected)
 
     def test_utf8_custom_data(self):
-        obj = self.portal['my-image']
-        obj.setDescription('áéíóú')
+        obj1 = self.portal['my-image']
+        obj1.setDescription('áéíóú')
+        obj2 = self.portal['my-image3']
```
Comment (djowett) [inline src/collective/cover/tests/test_textlinessortable_widget.py]: IIUC my-image2 will do, no?

File: src/collective/cover/tests/test_textlinessortable_widget.py [hunk 5]
```diff
@@ -158,20 +158,65 @@ def test_extract(self):
         self.assertDictEqual(extracted_value, expected)
 
     def test_utf8_custom_data(self):
```
Comment (djowett) [inline src/collective/cover/tests/test_textlinessortable_widget.py]: I was seeing the same error without customising the title, 
(in fact I found it difficult to reproduce when there WAS a custom title)
So for example, the following gives an error for me:

        title = u'El veloz murciélago hindú comía feliz cardillo y kiwi'
        obj = self.portal['my-image']
        obj.setTitle(title)
        obj.reindexObject()
        self.tile.populate_with_object(obj)

        self.assertIsInstance(self.tile.get_title(obj), unicode)
        self.assertEqual(self.tile.get_title(obj), title)

maybe this supplements what you've already done?

File: src/collective/cover/tests/test_textlinessortable_widget.py (hunk 4 above)
Comment (fulv) [reply src/collective/cover/tests/test_textlinessortable_widget.py]: IIUC, whether you use the title of the image object or set a custom title you will be hitting different code paths, which is why the existing tests did not catch this issue.  That is why I created a `my-image3` which has both a title and description containing accented letters.
I don't understand the robot test failures, though.
Maybe I'll try it without `my-image3` first to see if the failures go away.

File: src/collective/cover/widgets/textlinessortable.py (hunk 2 above)
Comment (fulv) [reply src/collective/cover/widgets/textlinessortable.py]: Yes, thanks, I'll change that.   I had a nagging suspicion.

**Comment by fulv** [comment]
Weird, that seems to have fixed the robot tests.  I have no idea why the robot tests depend on the `structure` GenericSetup test profile.

**Comment by djowett** [comment]
@fulv, It seems to be failing in the collection tile test and I wonder if it's something to do with the default ordering of the collection - which I don't know, but could be modification date & since you don't add a corresponding line [here](https://github.com/collective/collective.cover/blob/44da774bbd799a527a0fdb36c6983ef3d5eb8e0c/src/collective/cover/testing.py#L169) for image-3 - perhaps that becomes the "oldest" image in the collection, and therefore the first item matching the `.tile-header h1` selector [in the failing line of the robot test](https://github.com/collective/collective.cover/blob/44da774bbd799a527a0fdb36c6983ef3d5eb8e0c/src/collective/cover/tests/test_collection_tile.robot#L82)

Just a wild guess, HTH.  I'll approve the changes though

**Comment by hvelarde** [comment]
@fulv I need to make a release soon to include latest changes I made on the package yesterday; would you mind to fix this so we can merge it before?

**Comment by fulv** [comment]
Yes, I'll fix tomorrow.

**Comment by djowett** [comment]
Thanks guys! Good job!

**Comment by fulv** [comment]
thank you!

**Comment by hvelarde** [comment]
thanks to you both; release 1.6b3 is out since yesterday: https://pypi.python.org/pypi/collective.cover/1.6b3

**Pull request merged**